src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -365,6 +365,17 @@ pub(crate) fn get_tool_rustc_compiler(
     builder.compiler(target_compiler.stage.saturating_sub(1), builder.config.host_target)
 }
 
+/// Returns a compiler that is able to compile a `ToolTarget` tool for the given `target`.
+pub(crate) fn get_tool_target_compiler(builder: &Builder<'_>, target: TargetSelection) -> Compiler {
+    todo!("FIX");
+    if builder.host_target == target {
+        builder.compiler(0, builder.host_target)
+    } else {
+        // FIXME: should this be builder.top_stage to avoid rebuilds?
+        builder.compiler(1, target)
```

Comment:
Discussion: hm, for dist tools (cross-compiled for target platform), does the final distributed tool need to be built by the stage 2 compiler? Though I imagine the differences of tools produced by $\mathrm{rustc}\_{1}$ and $\mathrm{rustc}\_{2}$ are ABI-wise, but since they are not intended to be linked against nor depend on unstable features, that should be fine?

Reply:
Sorry, as this was a result of combining like 5 branches and many rebases, the step-by-step commits weren't as clean as I would like them to be. I was also discovering things as I went by the individual tools and what were their use-cases and then didn't rebase them back.